esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -502,6 +510,27 @@ void WUDetails::processRequest(IEspWUDetailsRequest &req, IEspWUDetailsResponse
                     espWuResponseNote->setErrorCode_null();
                 espWuResponseNote->setSeverity(cur.queryProp("@severity"));
                 espWuResponseNote->setCost(0);
+                VStringBuffer idstr("%" I64F "u", id|COMPONENT_NOTE_MASK);
```

Comment:
The id's here, being returned in setId (WuResponseNote::Id), they're 64-bit ..
For the non-static warnings, they will have been created by makeMessageId ? i.e. the bottom 21 bits correspond to the year/month/day, the id lies in the top bits (id<<21).
Is that correct?

If so, then the id's created here for static's should follow the same format (and use makeMessageId). Otherwise these 32-bit id's with the top-bit set will always decodeMessageId to an id of 0, and some bits set for the date.

And, I think COMPONENT_NOTE_MASK should move to sysinfologger.hpp, and be used as the upper limit to wrap on instead of the current UINT_MAX.

Reply:
Conclusion from offline chat is to introduce a setStaticMessageId, which sets a bit in the compound 64bit id, to indicate that it is static message id.